walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +359,49 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void noMandatoryJustOptional(){
+    // No mandatory attendees, just two optional attendees with several gaps in schedule.
+    //
+    // Events  : |--A--|  |--B--|  |-A-|
+    // Day     : |---------------------|
+    // Options :       |--|     |--|
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TIME_0930AM, false),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_1100AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_A)));
+    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList(TimeRange.fromStartEnd(TIME_0830AM, TIME_0900AM, false), TimeRange.fromStartEnd(TIME_0930AM, TIME_1100AM, false));
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noMandatoryJustOptionalNoGaps() {
+    // No mandatory attendees, just two optional attendees with no gaps in schedule.
+    //
+    // Events  : |--A-----|--B---------|
+    // Day     : |---------------------|
+    // Options :       
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0930AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true),
+            Arrays.asList(PERSON_B)));
+    MeetingRequest request = new MeetingRequest(Arrays.asList(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList();
+    Assert.assertEquals(expected, actual);
```

Comment:
go/unit-test-practices#structure

//Arrange

//Act

//Assert

with new lines in between.

```java
@Test
public void shouldReloadData() {
  when(myService.getData()).thenReturn("data1", "data2");

  myClass.initialize();
  myClass.loadData();
  myClass.loadData();
  String result = myClass.getData();

  assertThat(result).isEqualTo("data2");
}
```